clients/src/main/java/org/apache/kafka/common/requests/OffsetsForLeaderEpochRequest.java

```diff
@@ -105,13 +126,15 @@ public String toString() {
         }
     }
 
-    public OffsetsForLeaderEpochRequest(Map<TopicPartition, PartitionData> epochsByPartition, short version) {
+    public OffsetsForLeaderEpochRequest(Map<TopicPartition, PartitionData> epochsByPartition, int replicaId, short version) {
         super(ApiKeys.OFFSET_FOR_LEADER_EPOCH, version);
         this.epochsByPartition = epochsByPartition;
+        this.replicaId = replicaId;
     }
 
     public OffsetsForLeaderEpochRequest(Struct struct, short version) {
         super(ApiKeys.OFFSET_FOR_LEADER_EPOCH, version);
+        replicaId = struct.get(REPLICA_ID);
```

Comment:
Hmm this would fail for the old schema. I think we have to use `getOrElse`.

Reply:
Will do. I'll use the consumer sentinel value for the default